### PR TITLE
Fix spelling and add a check

### PR DESCRIPTION
**Pull request by asensio-project**
Hello,

This PR adds a GitHub Action for check typos, using the typos crate. It also adds some exceptions for the check and some spelling fixes.

Thanks!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -107,3 +107,11 @@ jobs:
...
       - name: Check semver
         run: cargo semver-checks check-release
+
+  typos:
+    name: Check typos
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: cargo install typos-cli --locked
+      - run: typos
```
Comment (phil-opp) [inline .github/workflows/ci.yml:117]: Any reason to not use the [GitHub action](https://github.com/crate-ci/typos/blob/master/docs/github-action.md) provided by that crate? E.g.:

```yaml
- name: Check spelling
  uses: crate-ci/typos@v1.0.4
```

I'm fine with the `cargo install` approach too if it works better.

**Review by phil-opp — COMMENTED** [review]
Great idea, thanks a lot!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (asensio-project) [reply .github/workflows/ci.yml:117]: I didn't think that, thanks! I will publish a new commit.

Comment (asensio-project) [reply .github/workflows/ci.yml]: I created the commit

Comment (asensio-project) [reply .github/workflows/ci.yml]: The spelling test passes!

Comment (phil-opp) [reply .github/workflows/ci.yml]: Great, thanks! Looks like it's quicker this way.

**Pull request merged**